src/sentry/snuba/json_schemas.py

```diff
@@ -29,8 +29,9 @@
                 "required": ["data"],
             },
             "timestamp": {"type": "string", "format": "date-time"},
+            "entity": {"type": "string", "minLength": 1},
         },
-        "required": ["subscription_id", "values", "timestamp"],
+        "required": ["subscription_id", "values", "timestamp", "entity"],
```

Comment:
Not sure how json schema works, but do we really want to add a key to and old schema version? Or rather create a new version for this?

Reply:
Why would we create a new schema?

Reply:
Oh I get what you mean. This is not metrics specific.. this applies to all subscription updates

Reply:
Yeah, we should be careful with backwards compatibility here. Usually a new `required` field would require a schema version bump.

Will all subscription updates have an entity now?

Reply:
@wedamija 
>Will all subscription updates have an entity now?

Yep 
How do we do a schema version bump?

Reply:
I also realised I should probably split this PR, have the json schema here alone, and only once the snuba PR is deployed then I extract the `entity` from the contents

Reply:
Moving schema to #30736

Reply:
Your order looks right, just to make sure though, are we already sending `entity` down in the updates? 

We should also consider what we do if we receive a schema v2 update with no entity, and maybe have backwards compat code in place for a couple of days.

Reply:
Yep https://github.com/getsentry/snuba/pull/2291/files#diff-f48e31c677eccff7bf14b091fe348640c394684c53c8d3a5a16d27e265d8b1abR53

> We should also consider what we do if we receive a schema v2 update with no entity, and maybe have backwards compat code in place for a couple of days

okay yeah that makes sense. I will handle that case as well

Reply:
Should be resolved